### PR TITLE
🛷 Add useTokenList

### PR DESCRIPTION
**Pull request by HubertNastaly**

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -36,9 +36,11 @@
   },
   "dependencies": {
     "@types/react": "17.0.1",
+    "@uniswap/token-lists": "^1.0.0-beta.27",
     "@web3-react/core": "6.1.1",
     "@web3-react/injected-connector": "6.0.7",
     "@web3-react/network-connector": "6.1.5",
+    "axios": "^0.24.0",
```
Comment (HubertNastaly) [inline packages/core/package.json:43]: Don't know if I could add axios or just use fetch

File: packages/core/src/hooks/useTokenList.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { TokenInfo } from '@uniswap/token-lists'
+import { ChainId } from '../constants'
+
+interface TokenList {
+  name: string
+  logoURI: string
+  tokens: TokenInfo[]
+}
+
+export function useTokenList(tokenListURI: string, chainId: ChainId = ChainId.Mainnet) {
```
Comment (tt-marek) [inline packages/core/src/hooks/useTokenList.ts:12]: I propose default chainId to be taken from currently connected network, that would be sth like this:
```ts
export function useTokenList(tokenListURI: string, overrideChainId: ChainId = null) {
  const {chainId} = useEthers()
  if (!overrideChainId) {
     overrideChainId = chainId
  } 

....
          tokens: (tokens as TokenInfo[]).filter((token) => token.chainId === overrideChainId),
...
  }, [tokenListURI, overrideChainId])
...
```

Comment (tt-marek) [reply packages/core/src/hooks/useTokenList.ts:12]: If chainId undefined/null - return empty list

File: packages/core/package.json (hunk 1 above)
Comment (tt-marek) [reply packages/core/package.json:43]: I would propose to use fetch indeed - less dependencies is better

**Comment by tt-marek** [comment]
I guess documentation is missing :)

File: packages/core/src/hooks/useTokenList.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          name,
+          logoURI,
+          tokens: (tokens as TokenInfo[]).filter((token) => token.chainId === chainId),
+        })
+      })
+      .catch((err) => {
+        console.log(err)
+        setTokenList(undefined)
+      })
+  }, [tokenListURI, chainId])
```
Comment (nezouse) [inline packages/core/src/hooks/useTokenList.ts:30]: Could just use plain fetch with checking if response is ok 
```js
.fetch(url)
    .then(async response => {
      if (response.ok) {
        return await response.json()
      } else {
        const errorMessage = await response.text()
        return Promise.reject(new Error(errorMessage))
      }
    }).catch(err)...
```

File: packages/example/src/components/TokensList/TokensList.tsx [hunk 4]
```diff
@@ -1,17 +1,15 @@
-import { formatUnits } from '@ethersproject/units'
-import uniswapToken from '@uniswap/default-token-list'
```
Comment (tt-marek) [inline packages/example/src/components/TokensList/TokensList.tsx:2]: Can we remove it now from package.json?

Comment (HubertNastaly) [reply packages/example/src/components/TokensList/TokensList.tsx:2]: ✅

File: packages/core/src/hooks/useTokenList.ts (hunk 3 above)
Comment (HubertNastaly) [reply packages/core/src/hooks/useTokenList.ts]: ✅

File: packages/core/src/hooks/useTokenList.ts (hunk 2 above)
Comment (HubertNastaly) [reply packages/core/src/hooks/useTokenList.ts]: ✅

File: packages/core/package.json (hunk 1 above)
Comment (HubertNastaly) [reply packages/core/package.json]: ✅

File: packages/example/src/utils/adjustImageSource.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export function adjustImageSource(src: string) {
```
Comment (tt-marek) [inline packages/example/src/utils/adjustImageSource.ts:1]: `adjustIPFSPath`

File: docs/source/core.rst [hunk 6]
```diff
@@ -442,6 +442,61 @@ Returns allowance (tokens left to use by spender) for given tokenOwner - spender
...
+
+If an error occurs ``undefined`` is returned.
+
+**Example**
+
+.. code-block:: javascript
+
+  const { name, logoURI, tokens } = useTokenList(UNISWAP_DEFAULT_TOKEN_LIST_URI) || {}
+
+  const httpSource = logoURI && logoURI.startsWith('ipfs') ? logoURI.replace('ipfs://', 'https://ipfs.io/ipfs/') : logoURI
```
Comment (HubertNastaly) [inline docs/source/core.rst:470]: Not sure if this should be there, but without it the example doesn't work and might be misleading

File: packages/example/src/utils/adjustImageSource.ts (hunk 5 above)
Comment (HubertNastaly) [reply packages/example/src/utils/adjustImageSource.ts]: ✅

File: packages/example/src/components/TokenList/TokenList.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  //             {token.logoURI && <TokenIcon src={token.logoURI} alt={`${token.symbol} logo`} />}
+  //           </TokenIconContainer>
+  //           <TokenName>{token.name}</TokenName>
+  //           <TokenTicker>{token.symbol}</TokenTicker>
+  //           {balances?.[idx] && <TokenBalance>{formatUnits(balances[idx]![0], token.decimals)}</TokenBalance>}
+  //         </TokenItem>
+  //       ))}
+  //   </List>
+  // )
+  const httpSource = logoURI && logoURI.startsWith('ipfs') ? logoURI.replace('ipfs://', 'https://ipfs.io/ipfs/') : logoURI
```
Comment (tt-marek) [inline packages/example/src/components/TokenList/TokenList.tsx]: Comment 🆘

File: docs/source/core.rst (hunk 6 above)
Comment (tt-marek) [reply docs/source/core.rst:470]: LGTM

**Review by tt-marek — APPROVED** [review]

**Review by tt-marek — APPROVED** [review]

**Pull request merged**